cluster/connections.go

```diff
@@ -49,9 +52,17 @@ func GetInstance() *Singleton {
 		//build connections cache
 		cnxCache := make(map[string]*sql.DB)
 
+		// Check if a connection to an external kms is available
+		kmsResult := <-connectToKms()
```

Comment:
I think we could instanciate this conneciton on demand, by placing it behind a `KmsClient()` function that starts the connection when it's needed, but maybe only for context